internal/dynamo/complex_keys.go

```diff
@@ -0,0 +1,248 @@
+package dynamo
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+type Keys struct {
+	PK PK
+	SK SK
+}
+
+func (k *Keys) read(pkStr, skStr string) error {
+	v, err := readKey(pkStr)
+	if err != nil {
+		return err
+	}
+
+	var ok bool
+	k.PK, ok = v.(PK)
+	if !ok {
+		return errors.New("newKeys pk not pk")
+	}
+
+	v, err = readKey(skStr)
+	if err != nil {
+		return err
+	}
+
+	k.SK, ok = v.(SK)
+	if !ok {
+		return errors.New("newKeys sk not sk")
+	}
+
+	return nil
+}
+
+func (k *Keys) UnmarshalDynamoDBAttributeValue(av types.AttributeValue) error {
+	var skeys struct{ PK, SK string }
+	if err := attributevalue.Unmarshal(av, &skeys); err != nil {
+		return err
+	}
+
+	return k.read(skeys.PK, skeys.SK)
+}
+
+func (k *Keys) UnmarshalJSON(text []byte) error {
+	var skeys struct{ PK, SK string }
+	if err := json.Unmarshal(text, &skeys); err != nil {
+		return err
+	}
+
+	return k.read(skeys.PK, skeys.SK)
+}
+
+type LpaOwnerKeyType struct{ sk SK }
+
+func LpaOwnerKey(sk interface {
```

Comment:
this was the best way I could think of dealing with the situations where we need to allow multiple keys for the same field, it is a bit clunky but mainly a problem that you only see in tests